src/firebaseApi.ts

```diff
@@ -0,0 +1,51 @@
+import * as api from "./api";
+
+const API_VERSION = "v1beta1";
+
+/**
+ * Represents the FirebaseProject resource returned from calling
+ * `projects.get` in Firebase Management API:
+ * https://firebase.google.com/docs/projects/api/reference/rest/v1beta1/projects#FirebaseProject
+ */
+export interface FirebaseProject {
+  projectId: string;
+  projectNumber: number;
+  displayName: string;
+  name: string;
+  resources: {
+    hostingSite: string;
+    realtimeDatabaseInstance: string;
+    storageBucket: string;
+    locationId: string;
+  };
+}
+
+export async function listProjects(
+  nextPageToken?: string,
+  projectsList?: FirebaseProject[]
+): Promise<FirebaseProject[]> {
+  let projects = projectsList || [];
```

Comment:
Above, you can use a default parameter to solve this problem:

```typescript
export async function listProjects(
  nextPageToken?: string,
  projectsList: FirebaseProject[] = []
): Promise<FirebaseProject[]> {
```

I believe you have to both remove the `?` and add ` = []` - `tsc` will complain with both. You are, after all, declaring that `projectsList` will always be defined :)

Reply:
This is much cleaner, thanks. Done.